### ISSUE TITLE
Multimode Gateway 2 won't OTA

### ISSUE THREAD

**Issue by wangyz1997** [AlexxIT/XiaomiGateway3#1379]
I bought a Multimode Gateway 2 the other day, and the stock firmware version was 1.0.3. I connected it to the intergration as soon as I received it, and everything just worked as expected. After I obtained the key, I decided to upgrade the firmware to version 1.0.7, as start with this version, Multimode Gateway 2 is possible to operate as subgateway of central gateway. However,  it failed immediately after I clicked the OTA button in MiHome app.

![45289f1330b6e74a7d461d393a233611](https://github.com/AlexxIT/XiaomiGateway3/assets/29337599/d2b137cc-481d-4b95-a1f6-649b4a1e8ffe)

I then restored the gateway to factory defaults, left intergration unconnected, and performed the OTA. This time the progress went on to 80%, got stuck, then failed.

![c22bef3d28be2f1edbbbd5afc3f80eaf](https://github.com/AlexxIT/XiaomiGateway3/assets/29337599/eaa49ffa-5157-4742-8c48-ae3aadc942b9)

I'm wondering if this has something to do with the intergration, or this is a MiHome issue? If this is an intergration issue, what should I do to restore the gateway and upgrade the firmware? Thank you for your nice intergration and looking forward to your reply.

**Comment by AlexxIT** [comment]
Disable integration. Reboot gateway. Update firmware. Enable integration.

**Comment by wangyz1997** [comment]
Thanks. I'll have a try now.

**Comment by wangyz1997** [comment]
I've just tried, and everything happened just like that I menthoned when I restored the gateway to factory settings: after I clicked the upgrade button, the blue led on gateway blinked, and the progress goes up to 80%, then stuck for several minutes, smart phone app returns to the "click to start the upgrade procedure" screen, just like nothing happened: no errors, no successful upgrade. Finally the blue led stopped blinking after another few minutes, the gateway worked as it was.

**Comment by wz69699** [comment]
Think you should ask xiaomi.

**Comment by wangyz1997** [comment]
Yes, I'm here to confirm if there is any relation between using the intergration and the issue. I'll then consult Xiaomi when I got the origin of this issue.

**Comment by AlexxIT** [comment]
This is know issue about updating to 1.0.7.
Because gateway flash Zigbee chip firmware. 
And integration may brake this process.

**Comment by wangyz1997** [comment]
Thanks for your reply. Is there any solution?

**Comment by AlexxIT** [comment]
https://github.com/AlexxIT/XiaomiGateway3/issues/1379#issuecomment-2165938931

**Comment by wangyz1997** [comment]
Yes, I followed your step, but it failed over and over again, the symptom is as mentioned earlier.

> I've just tried, and everything happened just like that I menthoned when I restored the gateway to factory settings: after I clicked the upgrade button, the blue led on gateway blinked, and the progress goes up to 80%, then stuck for several minutes, smart MiHome app returned to the "click to start the upgrade procedure" screen, just like nothing happened: no errors, no successful upgrades. Finally the blue led stopped blinking after another few minutes, MiHome showed the "upgrade failed screen", and the gateway worked as it was.

**Comment by wangyz1997** [comment]
Swapped to a new gateway from Xiaomi, OTA successed following steps from https://github.com/AlexxIT/XiaomiGateway3/issues/1379#issuecomment-2165938931 .
